File: Checkout.java
import java.util.Scanner;
import java.util.List;
import java.util.ArrayList;
import java.time.LocalDateTime;
public class Checkout {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String moreItems;
        	System.out.println("What is the customer's name?: ");
            String customerName = scanner.nextLine();
            String productName;
             double price;
             int quantity;
             double discountAmount;
	     int amountPaid;	
             List<String> allProduct =  new ArrayList<String>();
            List<Double> allPrices = new ArrayList<Double>();
            List<Integer> allQuantity = new ArrayList<Integer>();

        do {
            System.out.println("What did the user buy? ");
             productName = scanner.nextLine();
             allProduct.add(productName);

		System.out.println("How many pieces? ");
             quantity = scanner.nextInt();
             allQuantity.add(quantity);	

            System.out.println("How much per unit? ");
           
            while (!scanner.hasNextDouble()) {                
	 System.out.println("Invalid price. Please enter a number: ");
                scanner.next(); 
            }
            price = scanner.nextDouble();
            allPrices.add(price);
            
           

            System.out.println("Add more items? (yes/no): ");
            moreItems = scanner.next().toLowerCase();
            scanner.nextLine();

        } while (moreItems.equals("yes"));
	System.out.println("What is your name?: ");
            String name = scanner.nextLine();

		System.out.println("How much discount will he get?: ");
             discountAmount = scanner.nextDouble();
		
	

        
        System.out.println("SEMICOLON STORES");
System.out.println("MAIN BRANCH");
System.out.println("LOCATION: 312, HERBERT MACAULAY, SABO YABA, LAGOS.");
System.out.println("TEL: 07048840550");
System.out.println("Date:"+ LocalDateTime.now());
System.out.println("Cashier's Name" + " " + name);
System.out.println("Customer's Name" + " " + customerName);

System.out.println("==========================================================");
System.out.printf("%10s%10s%10s%10s\n", "ITEM", "QUANTITY", "PRICE",   "TOTAL" );
System.out.println("----------------------------------------------------------");

double sum = 0.0;
double discount = 0.0;
double VATamount = 0.0;
double billTotal = 0.0;

for(int index = 0; index <allProduct.size(); index++) {
    System.out.printf("%10s%10d%10.2f%10.2f\n", allProduct.get(index) , allQuantity.get(index), allPrices.get(index),   allPrices.get(index) * allQuantity.get(index));
    
    sum += allPrices.get(index) * allQuantity.get(index);
    discount = sum * discountAmount / 100;
	VATamount = sum * (17.50 / 100 );
    billTotal = sum - discount + VATamount;
}

System.out.println("----------------------------------------------------------");
System.out.printf("%10s%10s%10s\n", " ",  " ",  "SubTotal:" +" " +" " + " " +" " + " " +" " + sum);
System.out.printf("%10s%10s%10s\n", " ",  " ",  "Discount:" +" " +" " + " " +" " + " " +" " + discount);
System.out.printf("%10s%10s%10s\n", " ",  " ",  "VAT @ 17.50%:" +" " +" " + " " +" " + " " +" " + VATamount);

System.out.println("==========================================================");

System.out.printf("%10s%10s%10s\n", " ",  " ",  "Bill Total:" +"         " + billTotal);

System.out.println("==========================================================");

System.out.printf("%s%f%n", "THIS IS NOT A RECEIPT KINDLY PAY" + " ",     billTotal);

System.out.println("==========================================================");

System.out.println("How much did the customer give to you?"); 
 amountPaid = scanner.nextInt();

double balance = 0.0;

System.out.println("SEMICOLON STORES");
System.out.println("MAIN BRANCH");
System.out.println("LOCATION: 312, HERBERT MACAULAY, SABO YABA, LAGOS.");
System.out.println("TEL: 07048840550");
System.out.println("Date:"+ LocalDateTime.now());
System.out.println("Cashier's Name" + " " + name);
System.out.println("Customer's Name" + " " + customerName);

System.out.println("==========================================================");
System.out.printf("%10s%10s%10s%10s\n", "ITEM", "QUANTITY", "PRICE",   "TOTAL" );
System.out.println("----------------------------------------------------------");

for(int index = 0; index <allProduct.size(); index++) {
    System.out.printf("%10s%10d%10.2f%10.2f\n", allProduct.get(index) , allQuantity.get(index), allPrices.get(index),   allPrices.get(index) * allQuantity.get(index));
    
  	balance = amountPaid - billTotal;
}

System.out.println("----------------------------------------------------------");
System.out.printf("%10s%10s%10s\n", " ",  " ",  "SubTotal:" +" " +" " + " " +" " + " " +" " + sum);
System.out.printf("%10s%10s%10s\n", " ",  " ",  "Discount:" +" " +" " + " " +" " + " " +" " + discount);
System.out.printf("%10s%10s%10s\n", " ",  " ",  "VAT @ 17.50%:" +" " +" " + " " +" " + " " +" " + VATamount);

System.out.println("==========================================================");

System.out.printf("%10s%10s%10s\n", " ",  " ",  "Bill Total:" +"         " + billTotal);
System.out.printf("%10s%10s%10s\n", " ",  " ",  "Amount Paid:" +"         " + amountPaid);
System.out.printf("%10s%10s%10s\n", " ",  " ",  "Balance:" +"         " + balance);

System.out.println("==========================================================");

System.out.printf("%10s\n",  "       THANK YOU FOR YOUR PATRONAGE" );

System.out.println("==========================================================");


}
}

File: checkout.py
from datetime import datetime
import array as arr

customers_name = input("What is the customer's name?: " '\n')

all_product = []
all_prices = []
all_quantity = []


product = input("What did the user buy?: " '\n')
all_product.append(product)

quantity = int(input("How many pieces?: " '\n'))
all_quantity.append(quantity)

price = int(input("How much per unit?: " '\n'))
all_prices.append(price)

add = input("Add more items?: " '\n')

while (add == 'yes'):
	product = input("What did the user buy?: " '\n')
	all_product.append(product)

	quantity = int(input("How many pieces?: " '\n'))
	all_quantity.append(quantity)

	price = int(input("How much per unit?: " '\n'))
	all_prices.append(price)

	
	add = input("Add more items?: " '\n')

	if add == 'no':
		break

cashier_name = input("What is your name?: " '\n')

discount_given = int(input("How much discount will he get?: " '\n'))


print('SEMICOLON STORES')
print('MAIN BRANCH')
print('LOCATION: 312, HERBERT MACAULAY WAY, SABO YABA, LAGOS.')
print('TEL: 07048840550')
now = datetime.now()
print("Date:", now)
print("Cashier: ", cashier_name)
print("Customer Name: ", customers_name)

print('==============================================================')

print("    ", "ITEM",  "        ",  "QUANTITY",  "        ",  "PRICE", "        " 
, "TOTAL" )

print("--------------------------------------------------------------")


sum_total = 0
discount = 0
VAT_amount = 0
bill_total = 0

for index in range(len(all_product)):
    print(f"      {all_product[index]:10s}        {all_quantity[index]:10d}     {all_prices[index]:10.2f}   {all_prices[index] * all_quantity[index]:10.2f}")
    
 
sum_total += all_prices[index]   * all_quantity[index]
discount = sum_total * discount_given / 100
VAT_amount = sum_total * (17.50 / 100)
bill_total = sum_total - discount + VAT_amount


print("------------------------------------------------------------")
print("               ",      "                ",    "SubTotal: ",       "     ",            sum_total)
print("               ",      "                ",    "Discount: ",             "      ",        discount)
print("               ",      "                ",    "VAT @ 17.50: ",     "   ",       VAT_amount)
   
  
print('==============================================================')

print("               ",      "                ",    "Bill Total: ",     "     ",          bill_total)

print('==============================================================')

print("                    ",     " THIS IS NOT A RECEIPT KINDLY PAY  ",           bill_total)

print('==============================================================')

amount = int(input("How much did the customer give to you?: " '\n'))


print('SEMICOLON STORES')
print('MAIN BRANCH')
print('LOCATION: 312, HERBERT MACAULAY WAY, SABO YABA, LAGOS.')
print('TEL: 07048840550')
now = datetime.now()
print("Date:", now)
print("Cashier: ", cashier_name)
print("Customer Name: ", customers_name)

print('==============================================================')

print("    ", "ITEM",  "        ",  "QUANTITY",  "        ",  "PRICE", "        " 
, "TOTAL" )

print("--------------------------------------------------------------")

balance =  amount - bill_total



print("------------------------------------------------------------")
print("               ",      "                ",    "SubTotal: ",       "     ",            sum_total)
print("               ",      "                ",    "Discount: ",             "      ",        discount)
print("               ",      "                ",    "VAT @ 17.50: ",     "   ",       VAT_amount)
   
  
print('==============================================================')

print("               ",      "                ",    "Bill Total: ",     "     ",          bill_total)
print("               ",      "                ",    "Amount Paid: ",     "     ",          amount)
print("               ",      "                ",    "Balance: ",     "     ",          balance)

print('==============================================================')

print("         ", 'THANK FOR YOUR PATRONAGE',    "         ")

print('==============================================================')
